src/content_sao.cpp

```diff
@@ -1273,6 +1273,42 @@ std::string PlayerSAO::getWieldList() const
 	return "main";
 }
 
+ItemStack PlayerSAO::getWieldedItem() const
+{
+	const Inventory *inv = getInventory();
+	ItemStack ret;
+	if (inv) {
+		const InventoryList *mlist = inv->getList(getWieldList());
+		if (mlist && (getWieldIndex() < (s32)mlist->getSize()))
+			ret = mlist->getItem(getWieldIndex());
+		if (ret.name == "") {
+			const InventoryList *hlist = inv->getList("hand");
+			if (hlist)
+				ret = hlist->getItem(0);
+		}
+	}
+	return ret;
```

Comment:
Yeah, I could. But both versions are equally good readable and the change wouldn't improve anything.